yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnExtensionService.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler.cluster
+
+import java.io.Closeable
+import java.util.concurrent.atomic.AtomicBoolean
+
+import org.apache.hadoop.service.AbstractService
+import org.apache.hadoop.service.Service.STATE
+import org.apache.hadoop.util.ShutdownHookManager
```

Comment:
cutting out all the shutdown logic if it is not needed; shutdown hooks are ugly things anyway